[eufy] Changed two notification fields to boolean

joepadmiraal opened the pull request:
When using the latest version of the eufy-security-ws Docker container, the OpenHAB addon fails because it expects `NotificationRing` and `NotificationMotion` to be booleans now.

joepadmiraal committed:
[eufy] Changed two notification fields to boolean

itb3 reviewed (commented):
Thank you for your interest in this addon. Won't this change break functionality for people that havent updated to the latest version of the container? maybe we should have a converter that can handle both 0/1 and true/false for these fields, so it will be useable for more versions of the container?

joepadmiraal commented:
Hmm good point.
I tried to find in which version it has changed.
It seems like the properties have been added in schema v3:
https://github.com/bropat/eufy-security-ws/blob/master/docs/api_schema_versions.md

Now when I look at the source of the corresponding version (0.3.0) it seems like it was already a boolean:
https://github.com/bropat/eufy-security-ws/blob/0.3.0/src/lib/device/state.ts#L91

I looks like it should be a boolean, even in the older containers.
Or am I missing something?

itb3 commented:
I must admit it's a long time since I created this and currently I'm trying nodered, but that would mean it *never* worked? I guess the addon has worked at some point.
*edit* seems there is ringing and notificationRing, do you know what the difference is? From the source it seems ringing is the chennel to know if someone pushed the doorbell?

joepadmiraal commented:
I tried out v0.8.4 of eufy-security-ws as that was the active version around the time when you created this binding.
With that version it seems that the fields are indeed integers instead of booleans.
I'll check if I can make it work with both boolean and int's

joepadmiraal committed:
[eufy] Allow both boolean and int for the notification fields and implemented rings state event

joepadmiraal commented:
I've updated the PR.
The notification fields can now be integer or boolean.
For that I've created a BoolInt class, I don't know if this approach is what you had in mind?

I also added a rings state event as that's what I see when I ring the doorbell.
I mapped it to the `ringing` channel.

joepadmiraal committed:
[eufy] small cleanup

itb3 commented on bundles/org.openhab.binding.eufy/src/test/resources/state_ringing_changed.json:
```diff
@@ -0,0 +1,9 @@
+	"event": {
+		"source": "device",
+		"event": "rings",
+		"serialNumber": "T8140P8121432028",
```
I would take out / fake the serial number here because it's probably from your doorbell, better safe than sorry ;)

itb3 replied:
whoops, seems "I would not", have to take mine out as well...

itb3 reviewed (approved):
LGTM. thanks!

joepadmiraal committed:
[eufy] removed serial numbers

joepadmiraal commented:
Alright, removed all serials from the test json's.
I just made a new commit for it instead of rebasing as you would be able to find them anyways via git history.
